webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTable.js

```diff
@@ -39,55 +40,58 @@ ErrorAlerts.propTypes = {
   errors: PropTypes.arrayOf(PropTypes.string).isRequired,
 };
 
-const buildTableRows = (subscriptions, updatedQuantity) =>
-  subscriptions.results.map((subs) => {
-    if (updatedQuantity[subs.id]) {
-      return {
-        ...subs,
-        entitlementsChanged: true,
-        quantity: updatedQuantity[subs.id],
-        availableQuantity: subscriptions.availableQuantities[subs.id],
-      };
-    }
-    return {
-      ...subs,
-      availableQuantity: subscriptions.availableQuantities[subs.id],
-    };
-  });
-
-const buildPools = updatedQuantity =>
-  Object.entries(updatedQuantity)
-    .map(([id, quantity]) => ({
-      id,
-      quantity,
-    }));
-
 class SubscriptionsTable extends Component {
-  static getDerivedStateFromProps(nextProps, prevState) {
-    if (nextProps.subscriptions !== undefined) {
-      return {
-        rows: buildTableRows(
-          nextProps.subscriptions,
-          prevState.updatedQuantity,
-        ),
-      };
-    }
-    return null;
-  }
-
   constructor(props) {
     super(props);
+
     this.state = {
+      rows: undefined,
```

Comment:
do we actually need to set initial value as undefined?

Reply:
No, but since we compere it to `undefined` later, I think it make it more reasonable to understand.